src/include/simeng/arch/Architecture.hh

```diff
@@ -108,7 +108,8 @@ class Architecture {
 
   /** Updates System registers of any system-based timers. */
   virtual void updateSystemTimerRegisters(RegisterFileSet* regFile,
-                                          const uint64_t iterations) const = 0;
+                                          const uint64_t iterations,
+                                          const uint64_t retired) const = 0;
 };
```

Comment:
I have two thoughts on this: 1) the function now doesn't only update system timers so it should be renamed, 2) I can see this function growing and becoming messy in the future with args being required for all architectures but only used in one. i.e. `retired` is only needed by RISC-V but you must pass it in for AArch64 also.

If it could be made more agnostic somehow (maybe the function takes no arguments but then probes the core for the information it needs?) then that would be preferable

Reply:
The main reason I changed this was to play around with the control status registers, however I think the concept off what I am doing will persist e.g. updating "system" registers with information from the core. 1) should be an easy enough fix e.g. ```updateSytemRegisters``` or ```updateSystemRegistersFromCore```. 

2) becomes a bit harder, I can see your point but now have thought about the execution of it. I think the current function (and how it may expand in the future) could be seen as decently agnostic - from the core, pass in all the information it has and then the ISA can choose what it does with that information. Performance wise, I can't see the unused parameters having any noticeable impact.

The other suggested option is to pass the core object itself to the ISA and probe for information. This would require creating a new getter in the core per new value required (not terrible, but slightly more complicated than passing as a parameter). Currently the ISA has access to the register file set, passed as a parameter, and updates the register file itself. If the core were the sole parameter then there would be no access to the register file parameter without making it a public field which seems dangerous to me as the core now doesn't get to control how this is updated. Another option would be to pass both the register file set and the core object into the functions which seems a bit odd but I guess would work. Probing would theoretically be slower as you need to make a function call per value on every cycle (resp. to having access directly) although I don't think this will be particularly noticeable.

@FinnWilkinson @jj16791 Thoughts on how to proceed?

Reply:
We could pass the core object to a `updateSystemRegisters` function and then make the Archiecture class a Friend class of Core. This would allow the architecture to access protected and private member variables of Core simply by doing `core.piv_mem_var`

Reply:
Resolving as change removed